src/pathology-report/ReportComponent.resource.ts

```diff
@@ -1,28 +1,75 @@
+
 export interface EncounterResult {
+  patientName: { familyName: string; familyName2: string; givenName: string };
+  patientPhoneNumber: { value: string };
+  patientUuid: string;
   uuid: string;
   encounterDatetime: Date;
+  sampleStatusObs: { value: { display: string } };
+  referralStatusObs: { value: { display: string } };
+  sampleDropoffObs: { value: { display: string } };
+}
+export interface iPatient {
+  uuid: string;
+}
+export interface iConcept {
+  uuid: string;
+  name: string;
 }
 
 export async function getPatientEncounters(patientUuid): Promise<Array<EncounterResult>> {
   const searchPatientEncounters = await openmrsFetch(
-    `/ws/rest/v1/encounter?patient=c604dabc-2700-102b-80cb-0017a47871b2`,
+    `/ws/rest/v1/encounter?encounterType=${encounterTypeUuid}&patient=${patientUuid}`,
   );
   const fullPatientEncounterData = searchPatientEncounters.data.results.map(async (searchEnc) => {
-    const enc = await openmrsFetch(`/ws/rest/v1/encounter/${searchEnc.uuid}`);
-    return enc.data;
+    const enc = await openmrsFetch(`/ws/rest/v1/encounter/${searchEnc.uuid}?v=full`);
+    const searchPatient = await openmrsFetch(`/ws/rest/v1/patient/${patientUuid}?v=full`);
+
+    const retrievedObject = {
+      patientName: searchPatient.data.person.names[0],
+      patientPhoneNumber: searchPatient.data.person.attributes.filter(
+        (attribute) => attribute.attributeType.uuid == phoneNumberAttrUuid,
+      )[0],
+      patientHealthCenter: searchPatient.data.person.attributes.filter(
+        (attribute) => attribute.attributeType.uuid == healthCenterAttrTypeUuid,
+      )[0],
+      patientUuid: searchPatient.data.uuid,
+      encUuid: enc.data.uuid,
+      encounterDatetime: enc.data.encounterDatetime,
+      sampleStatusObs: enc.data.obs.filter((obs) => obs.concept.uuid === sampleStatusUuid)[0],
+      referralStatusObs: enc.data.obs.filter((obs) => obs.concept.uuid === referralStatusUuuid)[0],
+      sampleDropoffObs: enc.data.obs.filter((obs) => obs.concept.uuid === sampleDropOffUuid)[0],
+    };
+
+    return retrievedObject;
   });
-  // const data = await searchPatientEncounters.json();
-  // console.log("------------------" + data);
   return Promise.all(fullPatientEncounterData);
 }
 
-// export async function getPatients({location}){
-//     const searchPatientsResults = await openmrsFetch('/ws/rest/v1/patient?q=c604dabc-2700-102b-80cb-0017a47871b2',{
-//         method: 'GET',
-//     });
-//     const data = await searchResults.json();
-//     console.log("------------------" + data);
-//     return searchPatientsResults;
-// }
+export async function getConcept(conceptUuid) {
+  const concept = await openmrsFetch(`/ws/rest/v1/concept/${conceptUuid}?v=full`);
+  return concept.data.answers;
+}
+
+export async function getLocation(locationName) {
+  const location = await openmrsFetch(`/ws/rest/v1/location?q=${locationName}&v=default`);
+
+  return location.data.results;
```

Comment:
The function name is singular `getLocation` but this returns an array. Either call it `getLocations` or make it return a single item, like `location.data.results[0]`.